datalog/datalog.go

```diff
@@ -22,13 +22,30 @@ const (
 	IDTypeString
 	IDTypeDate
 	IDTypeBytes
+	IDTypeList
```

Comment:
Should we call this a set? We don't care about order or duplicate values.

Reply:
Can rename, but switching from `[]ID` to `map[ID]struct{}` representation is tricky, because of the `Bytes` ID that can't be a map key. It will compile but panic at runtime that it is unhashable, which isn't great from usage perspective. 
wdyt ?